src/warnet/server.py

```diff
@@ -322,51 +322,60 @@ def scenarios_available(self) -> list[tuple]:
             self.logger.error(msg)
             raise ServerError(message=msg) from e
 
-    def scenarios_run_file(
-        self, scenario_base64: str, additional_args: list[str], network: str = "warnet"
+    def _start_scenario(
+        self,
+        scenario_path: str,
+        scenario_name: str,
+        additional_args: list[str],
+        network: str,
     ) -> str:
-        scenario_path = None
-        with tempfile.NamedTemporaryFile(suffix=".py", delete=False) as temp_file:
-            scenario_path = temp_file.name
-
-            # decode base64 string to binary
-            scenario_bytes = base64.b64decode(scenario_base64)
-            # write binary to file
-            temp_file.write(scenario_bytes)
-
-        if not os.path.exists(scenario_path):
-            raise ServerError(f"Scenario not found at {scenario_path}.")
-
         try:
             run_cmd = [sys.executable, scenario_path] + additional_args + [f"--network={network}"]
             self.logger.debug(f"Running {run_cmd}")
-
             proc = subprocess.Popen(
                 run_cmd,
                 stdout=subprocess.PIPE,
                 stderr=subprocess.PIPE,
             )
-
             t = threading.Thread(target=lambda: self.proc_logger(proc))
             t.daemon = True
             t.start()
-
             self.running_scenarios.append(
                 {
                     "pid": proc.pid,
-                    "cmd": f"{scenario_path} {' '.join(additional_args)}",
+                    "cmd": f"{scenario_name} {' '.join(additional_args)}",
                     "proc": proc,
                     "network": network,
                 }
             )
-
-            return f"Running scenario with PID {proc.pid} in the background..."
-
+            return f"Running scenario {scenario_name} with PID {proc.pid} in the background..."
         except Exception as e:
             msg = f"Error running scenario: {e}"
             self.logger.error(msg)
             raise ServerError(message=msg) from e
 
+    def scenarios_run_file(
+        self,
+        scenario_base64: str,
+        scenario_name: str,
+        additional_args: list[str],
+        network: str = "warnet",
+    ) -> str:
+        # Extract just the filename without path and extension
+        base_name = os.path.splitext(os.path.basename(scenario_name))[0]
```

Comment:
I think it makes more sense to do this in cli/scenarios.py and just pass the name to the server. Could even theoretically be an option to add a custom `--name` when running a scenario. for example if you want to run the same scenario a few times in parallel.

`warcli scenarios stop` could even accept that name as an option instead of a pid !